src/frontend/js/components/ReferralTable/Input.tsx

```diff
@@ -1,18 +1,199 @@
+  QueryAutosuggestProps,
+  Suggestion,
+  SuggestionSection,
+} from './types';
 
 const messages = defineMessages({
   queryPlaceholder: {
     defaultMessage: 'Search in referrals',
     description: 'Placeholder for the query input in referral table.',
     id: 'components.ReferralTable.Input.queryPlaceholder',
   },
+  suggestionSectionAssignees: {
+    defaultMessage: 'Assignees',
+    description:
+      'Title for the suggestion section in the search autocomplete dropdown.',
+    id: 'components.ReferralTable.Input.suggestionSectionAssignees',
+  },
+  suggestionSectionTopics: {
+    defaultMessage: 'Topics',
+    description:
+      'Title for the suggestion section in the search autocomplete dropdown.',
+    id: 'components.ReferralTable.Input.suggestionSectionTopics',
+  },
+  suggestionSectionUnits: {
+    defaultMessage: 'Units',
+    description:
+      'Title for the suggestion section in the search autocomplete dropdown.',
+    id: 'components.ReferralTable.Input.suggestionSectionUnits',
+  },
+  suggestionSectionUsers: {
+    defaultMessage: 'Requesters',
+    description:
+      'Title for the suggestion section in the search autocomplete dropdown.',
+    id: 'components.ReferralTable.Input.suggestionSectionUsers',
+  },
 });
 
+const SUGGESTION_SECTIONS = [
+  {
+    column: FilterColumns.TOPIC,
+    kind: 'topiclites',
+    title: messages.suggestionSectionTopics,
+  },
+  {
+    column: FilterColumns.UNIT,
+    kind: 'unitlites',
+    title: messages.suggestionSectionUnits,
+  },
+  {
+    column: FilterColumns.USER,
+    kind: 'userlites',
+    title: messages.suggestionSectionUsers,
+  },
+  {
+    column: FilterColumns.ASSIGNEE,
+    kind: 'userlites',
+    title: messages.suggestionSectionAssignees,
+  },
+];
+
+/**
+ * `react-autosuggest` callback to get a human string value from a Suggestion object.
+ * @param suggestion The relevant suggestion object.
+ */
+export const getSuggestionValue: Autosuggest.GetSuggestionValue<Suggestion> = (
+  suggestion,
+) => {
+  switch (suggestion.kind) {
+    case 'topiclites':
+      return suggestion.value.name;
+    case 'unitlites':
+      return suggestion.value.name;
+    case 'userlites':
+      return getUserFullname(suggestion.value);
+  }
+};
+
+/**
+ * Build a suggestion section from a model name and a title, requesting the relevant
+ * values to populate it from the API
+ * @param kind The kind of suggestion we're issuing the completion request for. Determines the API
+ * endpoint we're sending the request to.
+ * @param title MessageDescriptor for the title of the section that displays the suggestions.
+ * @param query The actual payload to run the completion search with.
+ */
+export const getSuggestionsSection = async (
+  {
+    column,
+    kind,
+    title,
+  }: { column: FilterColumns; kind: string; title: MessageDescriptor },
+  query: string,
+) => {
+  // Run the search for the section on the API
+  let response: Response;
+  try {
+    response = await fetch(
```

Comment:
Maybe this call should be in /data folder ?

Reply:
IMO it makes sense to put it close to the input as it's very specific to the `Autosuggest` use case.